synapse/storage/events.py

```diff
@@ -596,10 +601,70 @@ def _get_events(txn, batch):
             )
 
         for chunk in batch_iter(event_ids, 100):
-            yield self.runInteraction("_get_events_which_are_prevs", _get_events, chunk)
+            yield self.runInteraction(
+                "_get_events_which_are_prevs", _get_events_which_are_prevs_txn,
+                chunk,
+            )
 
         defer.returnValue(results)
 
+    @defer.inlineCallbacks
+    def _get_prevs_before_rejected(self, event_ids):
+        """Given a set of events recursively find all prev events that have
```

Comment:
my understanding is that we start by checking the events themselves, rather than their prev events, for rejections. This description implies otherwise.

Also, the idea is that the first line is one-line description of the function (https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings). Can you try and make a one-liner, even if it's a bit opaque? ("Find transitive predecessors of some events, via rejections." ?)

Reply:
> my understanding is that we start by checking the events themselves, rather than their prev events, for rejections. This description implies otherwise.

So this starts with the events we're about to persist and checks their prev events, and their prev events, etc. This is the opposite to `_get_events_which_are_prevs` which works in the other direction. Since we haven't persisted the new event yet we have to pass in their prev events and start from there.